bundles/org.palladiosimulator.analyzer.workflow/src/org/palladiosimulator/analyzer/workflow/jobs/LoadMiddlewareConfigurationIntoBlackboardJob.java

```diff
@@ -1,92 +1,31 @@
 package org.palladiosimulator.analyzer.workflow.jobs;
 
-import org.apache.log4j.Logger;
-import org.eclipse.core.runtime.IProgressMonitor;
+import org.palladiosimulator.analyzer.workflow.ConstantsContainer;
 import org.palladiosimulator.analyzer.workflow.configurations.AbstractCodeGenerationWorkflowRunConfiguration;
-import org.palladiosimulator.analyzer.workflow.configurations.AbstractPCMWorkflowRunConfiguration;
 
-import de.uka.ipd.sdq.workflow.jobs.CleanupFailedException;
-import de.uka.ipd.sdq.workflow.jobs.IBlackboardInteractingJob;
-import de.uka.ipd.sdq.workflow.jobs.IJob;
-import de.uka.ipd.sdq.workflow.jobs.JobFailedException;
-import de.uka.ipd.sdq.workflow.jobs.UserCanceledException;
+import de.uka.ipd.sdq.workflow.jobs.SequentialBlackboardInteractingJob;
 import de.uka.ipd.sdq.workflow.mdsd.blackboard.MDSDBlackboard;
-import de.uka.ipd.sdq.workflow.mdsd.blackboard.ResourceSetPartition;
 
-public class LoadMiddlewareConfigurationIntoBlackboardJob 
-implements IJob, IBlackboardInteractingJob<MDSDBlackboard> {
+public class LoadMiddlewareConfigurationIntoBlackboardJob extends SequentialBlackboardInteractingJob<MDSDBlackboard> {
 
-    private static final Logger LOGGER = Logger.getLogger(LoadPCMModelsIntoBlackboardJob.class);
+    @Deprecated
+	public static final String RMI_MIDDLEWARE_PARTITION_ID = ConstantsContainer.RMI_MIDDLEWARE_REPOSITORY_PARTITION_ID;
 	
-	public static final String RMI_MIDDLEWARE_PARTITION_ID = "de.uka.ipd.sdq.pcmmodels.partition.rmimiddleware";
-	public static final String EVENT_MIDDLEWARE_PARTITION_ID = "de.uka.ipd.sdq.pcmmodels.partition.eventmiddleware";
-
-	private MDSDBlackboard blackboard = null;
-	private AbstractPCMWorkflowRunConfiguration configuration = null;
+	@Deprecated
+	public static final String EVENT_MIDDLEWARE_PARTITION_ID = ConstantsContainer.EVENT_MIDDLEWARE_REPOSITORY_PARTITION_ID;
 
 	public LoadMiddlewareConfigurationIntoBlackboardJob(AbstractCodeGenerationWorkflowRunConfiguration config) {
-		super();
-		
-		this.configuration  = config;
-	}
-
-	@Override
-    public void execute(IProgressMonitor monitor) throws JobFailedException,
-			UserCanceledException {
-		ResourceSetPartition middlewareRepositoryPartition = null;
-		if (!this.blackboard.hasPartition(RMI_MIDDLEWARE_PARTITION_ID)) {
-			if(LOGGER.isDebugEnabled()) {
-                LOGGER.debug("Creating RMI Middleware Repository Partition");
-            }
-			
-			middlewareRepositoryPartition = new ResourceSetPartition();
-			this.blackboard.addPartition(RMI_MIDDLEWARE_PARTITION_ID, middlewareRepositoryPartition);
-			
-			if(LOGGER.isDebugEnabled()) {
-                LOGGER.debug("Initialising RMI Middleware EPackages");
-            }
-			middlewareRepositoryPartition.initialiseResourceSetEPackages(AbstractPCMWorkflowRunConfiguration.PCM_EPACKAGES);

```

Comment:
(and line 67): is this initialization of the partition not required anymore? I would have expected it in the CreateBlackboardPartitionJob.

Reply:
If the job is run in eclipse (which is kind of an assumption, given that the job takes an Eclipse Progress Monitor), the registration is done by EMF automatically (via a generated extension in the model plugin). In case the job should be run in standalone, there are better ways to initialize the EPackages cache, than by hard-coding selected packaged.